acceptance/tests/allow_arbitrary_node_name_fact_for_agent.rb

```diff
@@ -11,9 +11,109 @@
 
 node_names.uniq!
 
-authfile = "#{testdir}/auth.conf"
-authconf = node_names.map do |node_name|
-  %Q[
+if @options[:is_puppetserver]
+  step "Ensure legacy auth.conf is disabled for test" do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => false}})
+  end
+
+  teardown do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
+  end
+
+  step "Backup the tk auth.conf file" do
+    on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.conf /etc/puppetlabs/puppetserver/conf.d/auth.bak'
```

Comment:
This is an exact copy of what we do here: https://github.com/puppetlabs/puppetserver/blob/master/acceptance/suites/tests/authorization/x509_auth.rb#L9-L16 . I can implement the behavior you've suggested but wanted to note that this would vary from the reference that I used.